src/renderer/pages/Staking/Overview/components/NominatorsList/ShardedList/ShardedList.tsx

```diff
@@ -0,0 +1,201 @@
+import { cnTw } from '@shared/lib/utils';
+import { FootnoteText, Checkbox, Accordion, Shimmering, Plate, Tooltip } from '@shared/ui';
+import { ShardAccount } from '@shared/core/types/account';
+import { DotStyles } from '@shared/ui/StatusLabel/common/constants';
+import { AssetBalance } from '@entities/asset';
+import { AssetFiatBalance } from '@entities/price/ui/AssetFiatBalance';
+import { useStakingData } from '@entities/staking';
+import { NominatorsItem } from '../NominatorItem/NominatorItem';
+
+type NominatorInfoWithShard = NominatorInfo<ShardAccount>;
+
+type Props = {
+  shardsStake: NominatorInfoWithShard[];
+  era?: number;
+  asset?: Asset;
+  explorers?: Explorer[];
+  isStakingLoading: boolean;
+  onCheckValidators: (stash?: Address) => void;
+  onToggleNominator: (nominator: Address, value?: boolean) => void;
+  getContent: (stake: NominatorInfoWithShard) => ReactNode;
+};
+
+export const ShardedList = ({
+  shardsStake,
+  era,
+  asset,
+  explorers,
+  isStakingLoading,
+  onCheckValidators,
+  onToggleNominator,
+  getContent,
+}: Props) => {
+  const { getNextUnstakingEra, hasRedeem } = useStakingData();
+
+  const { t } = useI18n();
+  const [shards, setShards] = useState<NominatorInfoWithShard[]>(shardsStake);
+  const [selectedShards, setSelectedShards] = useState<number>(0);
+  const [withdrawnShards, setWithdrawnShards] = useState<number>(0);
+  const [unstakingShards, setUnstakingShards] = useState<number>(0);
+
+  useEffect(() => {
+    setShards(shardsStake);
+  }, [shardsStake]);
+
+  const selectAllShards = (value: boolean) => {
+    const allShards = shards.map((shard) => {
+      onToggleNominator(shard.address, value);
+      setSelectedShards(value ? shards.length : 0);
+
+      return { ...shard, isSelected: value };
+    });
+
+    setShards(allShards);
+  };
+
+  const selectShard = (nominator: string, isChecked: boolean) => {
+    setSelectedShards((prev) => (isChecked ? prev + 1 : prev - 1));
+    onToggleNominator(nominator);
+  };
+
+  const getShards = (shardsStake: NominatorInfoWithShard[]) => {
+    if (!shards.length) return;
+
+    return shards.map((shard: NominatorInfoWithShard) => {
+      if (hasRedeem(shard.unlocking, era)) {
+        setWithdrawnShards((prev) => prev + 1);
+      }
+      if (getNextUnstakingEra(shard.unlocking, era)) {
+        setUnstakingShards((prev) => prev + 1);
+      }
+
+      return (
+        <li key={shard.account.id}>
+          <NominatorsItem
+            key={shard.account.id}
+            isStakingLoading={isStakingLoading}
+            content={getContent(shard)}
+            stake={shard}
+            nominatorsLength={shardsStake.length}
+            asset={asset}
+            explorers={explorers}
+            onToggleNominator={selectShard}
+            onCheckValidators={onCheckValidators}
+          />
+        </li>
+      );
+    });
+  };
+
+  const { totalReward, totalStake } = useMemo(() => {
```

Comment:
Usually `useMemo` is not worth the benefit, we create too many cache objects for not a heavy operation
`return acc`  doesn't create a new object on each iteration but `return { ... }` does

```suggestion
const { totalReward, totalStake } => {
  const total = shards.reduce((acc, shard) => {
    acc.totalReward += Number(shard.totalReward || 0);
    acc.totalStake += Number(shard.totalStake || 0);

    return acc;
  }, { totalReward: 0, totalStake: 0 });

  return {
    totalReward: total.totalReward.toString(),
    totalStake: total.totalStake.toString()
  };
};
```